### PR TITLE
Implementation of the function to add BBMASK

### PR DESCRIPTION
**Pull request by terasawa-lin**
# Purpose of pull request

setup-emlinux reads add_BBMASK.txt and writes it to conf/local.conf.

The following packages are listed in add_BBMASK.txt.
 poky/meta/recipes-bsp/grub/grub_2.02.bb
 poky/meta/recipes-bsp/grub/grub-efi_2.02.bb
 poky/meta/recipes-sato/webkit/webkitgtk_2.22.7.bb
 poky/meta/recipes-graphics/xorg-lib/libxxf86misc_1.0.4.bb
 meta-debian/recipes-debian/fcode-utils/fcode-utils_debian.bb

### CODE REVIEW

File: scripts/setup-emlinux [hunk 1]
```diff
@@ -55,6 +55,7 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
     # use poky's base-files package temporary.
     echo "BBMASK .= \"|meta-debian/recipes-debian/base-files/base-files_debian.bb\"" >> conf/local.conf
     echo "DL_DIR = \"\${TOPDIR}/../downloads\"" >> conf/local.conf
+    cat ../repos/meta-emlinux/scripts/add_BBMASK.txt >> conf/local.conf
```
Comment (hiraku-wfs) [inline scripts/setup-emlinux]: Please use ${REPOS} instead of "../repos".

File: scripts/add_BBMASK.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+BBMASK += "poky/meta/recipes-bsp/grub/grub_2.02.bb \
+           poky/meta/recipes-bsp/grub/grub-efi_2.02.bb \
+           poky/meta/recipes-sato/webkit/webkitgtk_2.22.7.bb \
+           poky/meta/recipes-graphics/xorg-lib/libxxf86misc_1.0.4.bb \
+           meta-debian/recipes-debian/fcode-utils/fcode-utils_debian.bb"
```
Comment (hiraku-wfs) [inline scripts/add_BBMASK.txt]: This file is not a program but a configuration file.
Please rename add_BBMASK.txt to bbmask.conf.

File: scripts/setup-emlinux [hunk 3]
```diff
@@ -55,6 +55,7 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
     # use poky's base-files package temporary.
     echo "BBMASK .= \"|meta-debian/recipes-debian/base-files/base-files_debian.bb\"" >> conf/local.conf
```
Comment (hiraku-wfs) [inline scripts/setup-emlinux]: Please include these mask configs in bbmask.conf.

**Review by hiraku-wfs — COMMENTED** [review]
This single commit does ...
- Separate BBMASK config from setup-emlinux
- Mask recipes which are unbuildable and unused in EMLinux

Please split it into two commits and add appropriate subjects to each commit.

**Comment by terasawa-lin** [comment]
Fixed.
Please let us know if you have any mistakes.

**Comment by terasawa-lin** [comment]
There was a mistake.

**Comment by terasawa-lin** [comment]
Fixed the mistake

File: scripts/setup-emlinux [hunk 4]
```diff
@@ -50,11 +50,10 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
     # Add config
     echo "DISTRO = \"emlinux\"" >> conf/local.conf
     echo "PACKAGE_CLASSES = \"package_deb\"" >> conf/local.conf
-    echo "BBMASK = \"meta-debian/recipes-core\"" >> conf/local.conf
     # nsswitch.conf is not provided by debian base-file package but systemd requires it.
     # use poky's base-files package temporary.
```
Comment (hiraku-wfs) [inline scripts/setup-emlinux]: Please move this comment to the bbmask.conf.

File: scripts/setup-emlinux [hunk 5]
```diff
@@ -50,11 +50,10 @@ if [ -z "$TEMPLATECONF" ] && [ "${EML_BUILDDIR_SETUP_DONE}" = "false" ]; then
...
-    echo "BBMASK .= \"|meta-debian/recipes-debian/base-files/base-files_debian.bb\"" >> conf/local.conf
     echo "DL_DIR = \"\${TOPDIR}/../downloads\"" >> conf/local.conf
+    cat ${REPOS}/meta-emlinux/scripts/bbmask.conf >> conf/local.conf
```
Comment (hiraku-wfs) [inline scripts/setup-emlinux:54]: This commit should create the bbmask.conf, which includes mask-configs about the recipes-core and the base-files.
In your next commit, please add the rest configs.

Comment (terasawa-lin) [reply scripts/setup-emlinux:54]: Is this comment as a commit message for setup-emlinux?

**Comment by terasawa-lin** [comment]
Fixed.

**Comment by hiraku-wfs** [comment]
Thanks!

**Pull request merged**